File: test/go/testing-matchers/main_test.go
package main

import (
	"cdk.tf/go/stack/generated/kreuzwerker/docker/config"
	"cdk.tf/go/stack/generated/kreuzwerker/docker/container"
	"cdk.tf/go/stack/generated/kreuzwerker/docker/datadockerimage"
	"cdk.tf/go/stack/generated/kreuzwerker/docker/datadockernetwork"
	"github.com/aws/jsii-runtime-go"
	"github.com/hashicorp/terraform-cdk-go/cdktf"
	"testing"
)

var app = cdktf.Testing_App(nil)
var stack = NewMyStack(app, "stack")
var synth = cdktf.Testing_Synth(stack, jsii.Bool(false))
var fullSynth = cdktf.Testing_FullSynth(stack)

var stackInvalid = NewMyStackInvalid(app, "stack-invalid")
var fullSynthInvalid = cdktf.Testing_FullSynth(stackInvalid)

func TestToHaveResourcePass(t *testing.T) {
	assertion := cdktf.Testing_ToHaveResource(synth, container.Container_TfResourceType())

	if !*assertion {
		t.Error("Assertion Failed")
	}
}

func TestToHaveResourceFail(t *testing.T) {
	assertion := cdktf.Testing_ToHaveResource(synth, config.Config_TfResourceType())

	if *assertion {
		t.Error("Assertion Failed")
	}
}

func TestToHaveResourceWithPropertiesPass(t *testing.T) {
	properties := map[string]interface{}{
		"ports": &[]*container.ContainerPorts{{
			Internal: jsii.Number(80),
			External: jsii.Number(8000),
		}},
	}
	assertion := cdktf.Testing_ToHaveResourceWithProperties(synth, container.Container_TfResourceType(), &properties)

	if !*assertion {
		t.Error("Assertion Failed")
	}
}

func TestToHaveResourceWithPropertiesFail(t *testing.T) {
	properties := map[string]interface{}{
		"ports": &[]*container.ContainerPorts{{
			Internal: jsii.Number(100),
			External: jsii.Number(1000),
		}},
	}
	assertion := cdktf.Testing_ToHaveResourceWithProperties(synth, container.Container_TfResourceType(), &properties)

	if *assertion {
		t.Error("Assertion Failed")
	}
}

func TestToHaveDataPass(t *testing.T) {
	assertion := cdktf.Testing_ToHaveDataSource(synth, datadockerimage.DataDockerImage_TfResourceType())

	if !*assertion {
		t.Error("Assertion Failed")
	}
}
func TestToHaveDataFail(t *testing.T) {
	assertion := cdktf.Testing_ToHaveDataSource(synth, datadockernetwork.DataDockerNetwork_TfResourceType())

	if *assertion {
		t.Error("Assertion Failed")
	}
}
func TestToHaveDataWithPropertiesPass(t *testing.T) {
	properties := map[string]interface{}{
		"name": "nginx:latest",
	}
	assertion := cdktf.Testing_ToHaveDataSourceWithProperties(synth, datadockerimage.DataDockerImage_TfResourceType(), &properties)

	if !*assertion {
		t.Error("Assertion Failed")
	}
}
func TestToHaveDataWithPropertiesFail(t *testing.T) {
	properties := map[string]interface{}{
		"name": "wrong",
	}
	assertion := cdktf.Testing_ToHaveDataSourceWithProperties(synth, datadockerimage.DataDockerImage_TfResourceType(), &properties)

	if *assertion {
		t.Error("Assertion Failed")
	}
}
func TestToBeValidTerraformPass(t *testing.T) {
	assertion := cdktf.Testing_ToBeValidTerraform(fullSynth)

	if !*assertion {
		t.Error("Assertion Failed")
	}
}

func TestToBeValidTerraformFail(t *testing.T) {
	assertion := cdktf.Testing_ToBeValidTerraform(fullSynthInvalid)

	if *assertion {
		t.Error("Assertion Failed")
	}
}

File: test/java/testing-matchers/MainTest.java
import com.hashicorp.cdktf.Testing;
import com.mycompany.app.Main;
import com.hashicorp.cdktf.App;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import imports.docker.container.Container;
import imports.docker.data_docker_image.DataDockerImage;
import imports.docker.data_docker_network.DataDockerNetwork;
import imports.docker.config.Config;
import java.util.HashMap;

public class MainTest {

    private final App app = Testing.app();
    private final Main main = new Main(app, "main-valid");
    private final Main.MainInvalid mainInvalidTerraform = new Main.MainInvalid(app, "main-invalid");
    private final String synthesized = Testing.synth(main);
    private final String fullSynthesized = Testing.fullSynth(main);
    private final String fullSynthesizedInvalidTerraform = Testing.fullSynth(mainInvalidTerraform);

    @Test
    void hasResourcePass() {
        assertTrue(Testing.toHaveResource(synthesized, Container.TF_RESOURCE_TYPE) );
    }

    @Test
    void hasResourceWithPropsPass() {
        assertTrue(Testing
                .toHaveResourceWithProperties(synthesized, Container.TF_RESOURCE_TYPE, new HashMap<String, Object>() {
                    {
                        put("dns_opts", new String[] { "1", "2", "3" });
                    }
                }) );
    }

    @Test
    void hasResourceFail() {
        assertFalse(Testing.toHaveResource(synthesized, Config.TF_RESOURCE_TYPE));
    }

    @Test
    void hasResourceWithPropsFail() {
        assertFalse(Testing
                .toHaveResourceWithProperties(synthesized, Container.TF_RESOURCE_TYPE, new HashMap<String, Object>() {
                    {
                        put("dns_opts", new String[] { "11", "22", "33" });
                    }
                }) );
    }

    @Test
    void hasDataPass() {
        assertTrue(Testing.toHaveDataSource(synthesized, DataDockerImage.TF_RESOURCE_TYPE));
    }

    @Test
    void hasDataWithPropsPass() {
        assertTrue(Testing.toHaveDataSourceWithProperties(synthesized, DataDockerImage.TF_RESOURCE_TYPE,
                new HashMap<String, Object>() {
                    {
                        put("name", "nginx:latest");
                    }
                }));
    }

    @Test
    void hasDataFail() {
        assertFalse(Testing.toHaveDataSource(synthesized, DataDockerNetwork.TF_RESOURCE_TYPE) );
    }

    @Test
    void hasDataWithPropsFail() {
        assertFalse(Testing.toHaveDataSourceWithProperties(synthesized, DataDockerImage.TF_RESOURCE_TYPE,
                new HashMap<String, Object>() {
                    {
                        put("name", "wrong");
                    }
                }) );
    }

    @Test
    void isValidTerrformPass() {
        assertTrue(Testing.toBeValidTerraform(fullSynthesized) );
    }

    @Test
    void isValidTerrformFail() {
        assertFalse(Testing.toBeValidTerraform(fullSynthesizedInvalidTerraform) );
    }

}